spec/rubocop/cop/lint/number_conversion_spec.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Lint::NumberConversion do
+  subject(:cop) { described_class.new(config) }
+
+  let(:config) { RuboCop::Config.new }
+
+  it 'registers an offense when using `#to_i`' do
+    expect_offense(<<-RUBY.strip_indent)
+      '10'.to_i
+      ^^^^^^^^^ Replace unsafe number conversion with number class parsing, instead of using "10".to_i, use stricter Integer("10").
+    RUBY
+  end
+
+  it 'registers an offense when using `#to_f`' do
+    expect_offense(<<-RUBY.strip_indent)
+      '10.2'.to_f
+      ^^^^^^^^^^^ Replace unsafe number conversion with number class parsing, instead of using "10".to_i, use stricter Integer("10").
+    RUBY
+  end
+
+  it 'registers an offense when using `#to_c`' do
+    expect_offense(<<-RUBY.strip_indent)
+      '10'.to_c
+      ^^^^^^^^^ Replace unsafe number conversion with number class parsing, instead of using "10".to_i, use stricter Integer("10").
```

Comment:
It would be better to indicate in a message to use `Complex('10')` instead of `'10'.to_c`.